### DISCUSSION TITLE
Solution: Geometry classes can't show on the screen, missing lines, etc

### DISCUSSION THREAD

**Discussion by niedong** [3b1b/manim#1546] [Show and tell]
Check your current graphics processor via `python -m moderngl`

This is my original settings:

```
moderngl 5.6.4
vendor: Intel
renderer: Intel(R) HD Graphics 620
version: 3.3.0 - Build 27.20.100.8854
python: 3.9.4 (tags/v3.9.4:1f2e308, Apr  6 2021, 13:40:21) [MSC v.1928 64 bit (AMD64)]
platform: win32
code: 330
```

During which I can't display any geometric classes(e.g. circle, square) and lines.

I changed my graphics processor via NVIDIA Control Panel(Please refer to your corresponding settings).

Now:

```
moderngl 5.6.4
vendor: NVIDIA Corporation
renderer: GeForce 940MX/PCIe/SSE2
version: 3.3.0 NVIDIA 461.40
python: 3.9.4 (tags/v3.9.4:1f2e308, Apr  6 2021, 13:40:21) [MSC v.1928 64 bit (AMD64)]
platform: win32
code: 330
```

And Everything becomes normal.

**Comment by enga018** [comment]
ok.. i must do the same.. or is there any other way that doesn't involve going out to market.

**Reply by fabriciosb** [reply]
You can try the community edition of manin, which does not use opengl yet: https://docs.manim.community/en/stable/
You can do mostly the same things, but the code you will write will be slight different (not compatible, overall, with 3b1b code).

**Reply by enga018** [reply]
what is their main difference..

**Reply by fabriciosb** [reply]
The 3b1b was developed by Grant Sanderson (the 3b1b guy) and was made open source. A group of people started the CE edition based on it, developing it further and makeing it more well documented. In the meantime, Grant also made various improvenments in his source code, so the projects have diverted. I don't really know which is better for what.

**Reply by enga018** [reply]
hi...in manimce version, 0.7.0 version

newbie here.  t2c doesnt work![IMG_20210701_230017.jpg](https://user-images.githubusercontent.com/1030201/124166106-1ffa3780-dac0-11eb-952a-f41ac91d28f8.jpg)

**Reply by fabriciosb** [reply]
I don't know what you mean by t2c. Anyway, I think this is not the proper channel for questions which are not related to bugs. You should check the discord chanel in: https://www.manim.community/ .

**Comment by ReleaseU** [comment]
Same issue with the Line class, it seems like the package cannot be applied with integrated intel graphic card.

**Comment by Sunlight-zero** [comment]
I am runing the system of Windows 11 and I cannot fix this bug by changing the settings in NVIDIA Control Panel. But I find that I can do that by changing the Windows 11 settings. Here is the method:

1. Open the Settings application.
2. Switch to System > Display > Graphics
3. Add Python.exe into the list (if not listed), press optional buttom of Python, and switch to "High performance"

Now python is using the GPU of NVIDIA and lines can be shown normally.
